src/tooling/tooling-engine-singleton.md

```diff
@@ -0,0 +1,123 @@
+<!--
+  ~ Copyright (c) godot-rust; Bromeon and contributors.
+  ~ This Source Code Form is subject to the terms of the Mozilla Public
+  ~ License, v. 2.0. If a copy of the MPL was not distributed with this
+  ~ file, You can obtain one at https://mozilla.org/MPL/2.0/.
+-->
+
+# Tooling with Engine Singletons
+
+It is important for you to understand the [Singleton (anti)-pattern](https://en.wikipedia.org/wiki/Singleton_pattern) to properly utilize this system.
+
+An Engine Singleton is registered through a specific Godot-provided Singleton.
+
+Engine Singletons in Godot:
+
+- are Object types
+- are always accessible to GDScript (and other bound langs like C#)
+- must be manually registered and unregistered in the `LevelInit::Scene` step
+
+
+## Defining a Singleton
+
+Defining a singleton is the same as the Simple Type for an Object Type.
+
+```rust
+#[derive(GodotClass)]
+#[class(tool, base=Object)]
+struct MyEditorSingleton {
+    #[base]
+    base : Base<Object>
+}
+
+#[godot_api]
+impl IObject for MyEditorSingleton {
+    fn init(base : Base<Object>) -> Self {
+
+        // you may want to omit this by adding "init" to the #[class(...)]
+        Self {
+            base
+        }
+    }
+}
+
+#[godot_api]
+impl MyEditorSingleton {
+
+    #[func]
+    fn foo(&mut self) {}
+
+}
+```
+
+
+## Registering a singleton
+
+Registering singletons is done during the `InitLevel::Scene` stage of initialization.
+
+
+```rust
+struct MyExtension;
+
+unsafe impl ExtensionLibrary for MyExtension {
+
+    // these functions are optional if you are not using an 
+    /// Engine Singleton or similarly manually registred types
+    fn on_level_init(level: InitLevel) {
```

Comment:
This comment is not necessary, as the whole point of this page is registering a singleton.

Maybe instead you can add to the initial text (prose, not code block) something like:

> To achieve this, we can customize our init/shutdown routines by overriding `ExtensionLibrary` trait methods.